webapp/src/app/components/sidenav/sidenav.component.ts

```diff
@@ -28,8 +32,21 @@ export class SidenavComponent implements OnInit {
 				this.selectedLayer = layer;
 			}
 		});
-		this.mapLoader.tileMap.subscribe(tilemap => this.tilemap = tilemap);
-		this.globalEvents.currentView.subscribe(view => this.currentView = view);
+		this.mapLoader.tileMap.subscribe(tilemap => {
+			this.tilemap = tilemap;
+			this.globalEvents.currentView.next(EditorView.Layers);
+		});
+		this.globalEvents.currentView.subscribe(view => {
+			this.currentView = view;
+			switch (view) {
```

Comment:
what could be simplified? `tabChanged` only gets called if the tab changes, so `activeTab` has to be set here

Reply:
Right now `tabChanges` emits this event to be subscribed in the same class. This mode can easily be moved to `tabChanged` to reduce complexity. I don't know if this event is used in another place as well but if it isn't it can be removed entirely.

Reply:
It is used when loading a new map to always switch to the layers tab

Reply:
I'm not really sure now. The view can be changed outside by setting a new value to `currentView`. Is here still something that needs to be refactored then?

Reply:
Well, if that's the case then it needs to be there